blockchain/blockchain.go

```diff
@@ -2128,8 +2141,17 @@ func (bc *BlockChain) ApplyTransaction(chainConfig *params.ChainConfig, author *
 		return nil, 0, err
 	}
 	if vmConfig.EnableInternalTxTracing {
-		// TODO-InternalTxTracer
-		// Need to store traces to the database or send them to the designated channel
+		switch tracer := vmConfig.Tracer.(type) {
+		case *vm.InternalTxTracer:
+			result, err := tracer.GetResult()
+			if err != nil {
+				logger.Error("failed to get tracing result from a transaction", "txHash", tx.Hash().String(), "err", err)
+				return nil, 0, err
+			}
+			bc.internalTraces = append(bc.internalTraces, result)
+		default:
+			logger.Warn("internal transaction tracing is on, but the VM tracer type is not the expected one", "type", reflect.TypeOf(tracer).String())
```

Comment:
How about changing the comment like below?
```suggestion
			logger.Warn("To trace internal transactions, VM tracer type should be vm.InternalTxTracer", "actualType", reflect.TypeOf(tracer).String())
```

Reply:
Applied! 5a0bc53962a26e25a7d73be8083f76d3a19f4a3c